qemu/tests/boot_order_check.py

```diff
@@ -73,17 +74,21 @@ def run(test, params, env):
     boot_fail_infos = boot_fail_infos % (list_nic_addr[0][0],
                                          list_nic_addr[1][0],
                                          list_nic_addr[2][0])
-
-    error.context("Check the guest boot result", logging.info)
     if bootorder_type == "type2":
-        session_serial = vm.wait_for_serial_login(timeout=timeout)
-        output = vm.serial_console.get_stripped_output()
-        session_serial.close()
+        chk_str = "Booting from Hard Disk."
     else:
+        chk_str = "No bootable device."
```

Comment:
Is there some reason to have to do a pre-check here? It isn't good because

1. it is not necessary since we can go to check `boot_fail_infos` directly.
2. have to achieve that by hardcoding.

Reply:
Thanks. I think the main reason to do this is to separate the `test.fail` from `test.error`, if we directly check the L99 in a loop, the results we can get are only `PASS` or `test.error(timeout)`, we will miss the `test.fail` if we do get the whole serial log but just mismatch the `boot_fail_infos` pattern.